core/src/main/java/com/opensymphony/xwork2/ObjectFactory.java

```diff
@@ -188,11 +188,7 @@ public Object buildBean(String className, Map<String, Object> extraContext) thro
      */
     public Object buildBean(String className, Map<String, Object> extraContext, boolean injectInternal) throws Exception {
         Class clazz = getClassInstance(className);
-        Object obj = buildBean(clazz, extraContext);
-        if (injectInternal) {
```

Comment:
But this is a part of the interface and this boolean is used in other implementations. And Spring version won't inject internal Struts dependencies without this flag.

Reply:
Yes, by refactoring I meant also in sub classes. It looks like not a burden to refactor all :) Elsewhere, I feel your improvement won't work when user don't use Struts core default object factory i.e. uses spring, cdi, osgi or plexus plugins for example. You already removed it's usage in spring in this PR. At a bottom, personally I feel it strange that a superclass expose something for subclasses that it itself don't need and use at all.

Reply:
`container.inject(clazz)` will do the same, I mean it will instantiate an object and inject all the internal dependencies. Those won't happen when using Spring or CDI, that's why leaving this up to the given factory is a good idea.

There is at least one implementation out of the Apache Struts control - Guice connector - and refactoring those methods can affect this plugin. We can annotate it as deprecated but I'm not sure if this is a good approach anyway.